OneSchedule.Domain/NotificationScheduler.cs

```diff
@@ -0,0 +1,39 @@
+﻿using OneSchedule.Domain.Abstractions;
+using OneSchedule.Entities;
+using Quartz;
+using Quartz.Impl;
+using Quartz.Lambda;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Telegram.Bot;
+
+namespace OneSchedule.Domain
+{
+    public class NotificationScheduler : INotificationScheduler
+    {
+        private readonly ITelegramBotClient _bot;
+        IScheduler _scheduler;
+
+        public NotificationScheduler(ITelegramBotClient bot, IScheduler scheduler)
+        {
+            _bot = bot;
+            _scheduler = scheduler;
+        }
+
+        public async void ScheduleNotifications(EventEntity eventEntity)
+        {
+            var tasks = new List<Task>();
```

Comment:
Here you create a List with default size. And in case if you have a lot of notifications to set, when you add new tasks to list, list is resized, elements from existing list are copied to the bigger one and it slows down the app. You can specify in constructor the exact size of the list you need